src/ompl/base/objectives/src/PathLengthOptimizationObjective.cpp

```diff
@@ -62,10 +64,13 @@ ompl::base::Cost ompl::base::PathLengthOptimizationObjective::motionCostHeuristi
     return motionCost(s1, s2);
 }
 
-#if OMPL_HAVE_EIGEN3
 ompl::base::InformedSamplerPtr ompl::base::PathLengthOptimizationObjective::allocInformedStateSampler(const ProblemDefinitionPtr probDefn, unsigned int maxNumberCalls) const
 {
-    // Make the direct path-length informed sampler and return
+    // Make the direct path-length informed sampler and return. If OMPL was compiled with Eigen, a direct version is available, if not a rejection-based technique can be used
+#if OMPL_HAVE_EIGEN3
     return boost::make_shared<PathLengthDirectInfSampler>(probDefn, maxNumberCalls);
-}
+#else
+    OMPL_INFORM("%s: Direct sampling of the path-length objective requires Eigen. As this version of OMPL was compiled without Eigen support, the informed sampling is defaulting to rejection sampling.", description_.c_str());
+    return boost::make_shared<RejectionInfSampler>(probDefn, maxNumberCalls);
```

Comment:
Hm, still thinking this through.
I guess if in the absence of Eigen `PathLengthOptimizationObjective::allocInformedStateSampler` throws, a user could still create a rejection sampler manually for path length.

I.e. instead of,

    InformedSamplerPtr infSampler = opt_->allocInformedSampler(...);

doing

    InformedSamplerPtr infSamplerboost::make_shared<RejectionInfSampler>(...);

I guess that leaves all the functionality available to a motivated user. The only problematic case I can think of would be in the InformedStateSampler wrapper class, which calls `opt_->allocInformedSampler(...)` by default, but an overloaded constructor could look after that?...